lightning/src/routing/router.rs

```diff
@@ -2398,6 +2398,9 @@ where L::Target: Logger {
 		// because we deterministically terminated the search due to low liquidity.
 		if !found_new_path && channel_saturation_pow_half != 0 {
 			channel_saturation_pow_half = 0;
+		} else if already_collected_value_msat < final_value_msat && path_value_msat != recommended_value_msat && !found_new_path {
+			log_trace!(logger, "Failed to collect enough value, but running again to collect extra paths with a potentially higher limit.");
+			path_value_msat = recommended_value_msat;
```

Comment:
Should we only do this if we `hit_minimum_limit`? It doesn't hurt to go around again, but isn't `hit_min_limit` the only way for there to be uncollected paths remaining?